xdsl/ir/core.py

```diff
@@ -967,6 +967,13 @@ def walk(
         if region_first:
             yield self
 
+    def walk_blocks_preorder(self) -> Iterator[Block]:
+        for region in self.regions:
+            for block in region.blocks:
```

Comment:
This is not what I would have expected a preorder traversal of blocks to do. Does llvm simply iterate over them in the order they appear?

Reply:
Now that I think about it, it would probably be worth adding a `walk` method on `Block` that mirrors the API of the Operation walk, and to change this to forward to that one. I also prefer our `walk_regions_first` to `Preorder` as it's more declarative, maybe for blocks it could be `walk_child_blocks_first`? pre-post order doesn't make sense in the absence of left and right children

Reply:
I more meant that I would have expected a "traversal" to explore blocks via a depth first search of successors from each block, rather than just return each block in order

Reply:
Ah no that's not what I would expect TBH. @gabrielrodcanal, if you don't feel like implementing reversed iteration and post-order then I think it would still be worth adding `walk` on Block, and `walk_blocks` on Operation, and we can add the optional parameters later without API breaking.